src/dependency-manager/src/utils/slugs.ts

```diff
@@ -0,0 +1,270 @@
+
+export type ArchitectSlug = string; // simply string that passes: ArchitectSlugValidator (ie "account-name")
+export type ComponentSlug = string; // "account-name/component-name"
+export type ComponentTag = string; // "tag"
+export type ComponentVersionSlug = string; // "account-name/component-name:tag"
+export type ServiceSlug = string; // "account-name/component-name/service-name"
+export type ServiceVersionSlug = string; // "account-name/component-name/service-name:tag"
+export type EnvironmentSlug = string; // "@environment-account-name/environment-name"
+export type ServiceInstanceSlug = string; // "account-name/component-name/service-name:tag@environment-account-name/environment-name"
+export type GatewaySlug = 'gateway'; // the string-literal
+export type InterfacesSlug = string; // <account-name>/<component-name>:<tag>-interfaces
+
+export type SlugKind = 'component' | 'component_version' | 'service' | 'service_version' | 'environment' | 'service_instance' | 'gateway' | 'interfaces';
+
+/**
+ * Represents the unique elements of a fully parsed slug. This object will have different fields depending on the
+ * type of slug (ie was it a tagged component vs an untagged component vs a service slug)
+ */
+export interface ParsedSlug {
```

Comment:
Do we still need these interfaces? Seems heavy - especially if we want to be able to sanitize any string value

Reply:
Are these only used in the case of a splitter?

Reply:
Copying this from offline from slack for posterity:

yes, those are used as the return values of the splitters. I think they will be useful. That is what saves us from having to do:

`ref.split('/')[2]` when we want to get at the service_name say.
Instead, it's: `const { service_name} = Slugs.splitServiceSlug(ref);` which feels both more typesafe and readable